dwave/system/samplers/dwave_sampler.py

```diff
@@ -345,14 +346,13 @@ def sample(self, bqm, warnings=None, **kwargs):
 
         solver = self.solver
 
-        if not (solver.nodes.issuperset(bqm.linear) and
-                solver.edges.issuperset(bqm.quadratic)):
+        try:
+            future = solver.sample_bqm(bqm, **kwargs)
+        except ProblemStructureError:
             msg = ("Problem graph incompatible with solver. Please use 'EmbeddingComposite' "
                    "to map the problem graph to the solver.")
             raise BinaryQuadraticModelStructureError(msg)
```

Comment:
do we maybe want to use a `except ProblemStructureError as err` and `from err` here?

Reply:
sure, makes sense